### PR TITLE
Update conflict resolution

### PR DESCRIPTION
**Pull request by aarondr77**
# Changes

This attempts to fix the following bug: 
- User 1 merges.
- User 2 merges a conflict, doesn’t resolve yet.
- User 3 merges, non conflicting with everyone.
- User 2 resolves conflicts.
- User 3’s work is lost.

It does not break any tests, so I will leave it here, but I also don't think it fixes the intended behavior. It is quite difficult to test on one computer -- Excel seems to crash sometimes when trying to run saga on 3 workbooks loaded to the same project. Let's test it during our next round of testing. 

- [x ] I ran all the tests to make sure I didn't break anything.

### CODE REVIEW

**Comment by naterush** [comment]
The following bug exists:
1. User 1 makes project, shares with 2.
2. User 2 changes A1 to 1 and merges.
3. User 1 changes A1 to 2, and merges, and gets conflicts, but does not resolve them yet.
4. User 2 changes B1 to 1, and merges.
5. User 1 resolves conflicts, and clicks submit. 

However, in this case, User 2's changes to B1 will be lose. The merge at the end of conflict resolution appears to not actually be merging correctly.

When we discussed this PR, we said that a commit on master would happen at the end of merge would occur if and only if there was no merge conflicts, but a commit is happening here either way. I'm not sure if this is why the bug exists, but at some point I thought it was important to have this constraint.

File: src/saga/merge.js [hunk 1]
```diff
@@ -60,9 +60,11 @@ async function resolveMergeConflicts(context, resolutions) {
         } 
     }
 
+    // make resolution commit on master
     await commit(context, "resolved merge conflicts", "resolved merge conflicts", "master");
 
-    return {status: mergeState.MERGE_SUCCESS, conflicts: null};
+    // update the shared version
+    return runMerge([]);
```
Comment (naterush) [inline src/saga/merge.js]: This is a bad practice and possibly will lead to bugs. Running an operation from inside another operation (e.g. you already have a context here) means that sync will turn back on while your still inside the outer run call.

I've gotten really weird, unpredictable behavior by doing this by accident (not even just from sync, but also from contexts), so I would change this to a direct call to merge.

**Review by naterush — CHANGES_REQUESTED** [review]
This doesn't fix an even simper 2-user version of the lost work bug. I think it's b/c of the always commit on master which I think should be conditional on no merge conflicts, but not totally sure.

**Comment by aarondr77** [comment]
@naterush this is fixed now. The main issue was that when I switched to running merge at the end of conflict resolution, I did not also switch the commit at the end of the conflict resolution to a commit on personal. Instead, it was committing the personal sheets directly to master, which overwrote the new changes. 

lmk if you have questions.

**Comment by naterush** [comment]
@aarondr77 the code looks good to me, but it's not working. I'm still loosing work. To replicate:
1. User 1 makes project, shares with 2.
2. User 2 changes A1 to 1 and merges.
3. User 1 changes A1 to 2, and merges, and gets conflicts, but does not resolve them yet.
4. User 2 changes B1 to 1, and merges.
5. User 1 resolves conflicts, and clicks submit.

User 2's changes to B1 disappear.

**Comment by naterush** [comment]
I'm pretty confused. It appears a merge commit is still occurring on master even when there is a merge conflict. 
Screen Shot 2020-05-28 at 6 28 57 PM

@aarondr77 this is certainly not intended behavior... any thoughts?

**Comment by aarondr77** [comment]
Hmmm I just reran that scenario and it worked on my end. See photos: 
Screen Shot 2020-05-28 at 7 26 23 PM
Screen Shot 2020-05-28 at 7 27 25 PM

**Comment by aarondr77** [comment]
Are you running the most up to date code? you pull?

**Comment by naterush** [comment]
Looks good. I was running the wrong code.

**Pull request merged**